Passes status of scenario and failed reason

AFASbart opened the pull request:
For the flutter_gherkin package I have added the possibility to pass along the scenario's final status and to retry a specific step multiple times.

AFASbart commented:
@jonsamwell Do you have the time to check this PR so we can resolve 3 (!) issues on the flutter_gherkin branch?

jonsamwell commented on lib/src/configuration.dart:
```diff
@@ -85,6 +85,8 @@ class TestConfiguration {
     this.createWorld,
   });
 
+  int stepMaxRetries = 3;
```
We should add documentation to this parameter.  Just something short.  I actually think this should default to zero to preserve existing behaviour.  It is actually bad practice to retry.  With cucumberjs to have to be explicit about the number of retries for a step / scenario

jonsamwell commented on lib/src/gherkin/steps/step_definition.dart:
```diff
@@ -60,19 +60,16 @@ abstract class StepDefinitionGeneric<TWorld extends World> {
-        st,
-      );
+    } on Error catch (e, st) {
+      return ErroredStepResult(elapsedMilliseconds, StepExecutionResult.error,
```
Add trailing comma and new line

AFASbart commented:
Before merging, can you recheck the flutter PR: https://github.com/jonsamwell/flutter_gherkin/pull/221. The max retries need to be atleast 1 ofcourse (so I added a + 1 on the for loop) and I have added a delay after each failed attempt.
